include/exadg/vector_tools/arbitrary_solution_transfer.h

```diff
@@ -0,0 +1,164 @@
+               dof_handler_src_in.get_triangulation(),
+               mapping_src_in);
+  }
+
+  /**
+   * Transfer the solution between source and target triangulations, that were set with reinit.
+   * At support points which are not overlapping the source triangulation, corresponding dof entries
+   * are set to 0.
+   *
+   * @param[in] dst Target Dof Vector.
+   * @param[in] src Source Dof Vector.
+   */
+  template<int n_components, typename VectorType1, typename VectorType2>
+  void
+  transfer_solution(VectorType1 &                                               dst,
+                    VectorType2 const &                                         src,
+                    dealii::VectorTools::EvaluationFlags::EvaluationFlags const flags =
+                      dealii::VectorTools::EvaluationFlags::avg) const
+  {
+    src.update_ghost_values();
+    auto const values =
+      dealii::VectorTools::point_values<n_components>(rpe, *dof_handler_src, src, flags);
+    src.zero_out_ghost_values();
+
+    fill_dof_vector_with_values<n_components>(dst, *dof_handler_dst, values);
+  }
+
+private:
+  std::vector<dealii::Point<dim>>
+  collect_mapped_support_points(dealii::DoFHandler<dim> const & dof_handler,
+                                dealii::Mapping<dim> const &    mapping) const
+  {
+    std::vector<dealii::Point<dim>> support_points;
+
+    for(auto const & cell : dof_handler.active_cell_iterators())
+    {
+      if(cell->is_locally_owned() == false)
+        continue;
+
+      auto cellwise_support_points = cell->get_fe().get_generalized_support_points();
+
+      for(auto & csp : cellwise_support_points)
+        csp = mapping.transform_unit_to_real_cell(cell, csp);
+
+      support_points.insert(support_points.end(),
+                            cellwise_support_points.begin(),
+                            cellwise_support_points.end());
+    }
+
+    return support_points;
+  }
+
+  template<int n_components, typename VectorType, typename ValueType>
+  void
+  fill_dof_vector_with_values(VectorType &                    dst,
+                              dealii::DoFHandler<dim> const & dof_handler,
+                              std::vector<ValueType> const &  values) const
+  {
+    auto ptr = values.begin();
+    for(auto const & cell : dof_handler.active_cell_iterators())
+    {
+      if(cell->is_locally_owned() == false)
+        continue;
+
+      unsigned int const n_support_points = cell->get_fe().get_generalized_support_points().size();
+
+      if constexpr(n_components == 1)
+      {
+        cell->set_dof_values(
+          dealii::Vector<typename VectorType::value_type>(ptr, ptr + n_support_points), dst);
+      }
+      else
+      {
+        cell->set_dof_values(multiple_components_to_dof_values<typename VectorType::value_type>(
+                               cell->get_fe(), n_support_points, ptr),
+                             dst);
+      }
```

Comment:
It looks like the same deal.II function (`set_dof_values()`) is called in both cases? If so, I would prefer that the vector `dof_values` is filled depending on the parameter `n_components`, and that the function call appears only once after the vector with `dof_values` has been filled.